packages/server/src/utilities/accessLevels.js

```diff
@@ -1,100 +1,36 @@
-const viewController = require("../api/controllers/view")
-const modelController = require("../api/controllers/model")
-const workflowController = require("../api/controllers/workflow")
-
-// Access Level IDs
-const ADMIN_LEVEL_ID = "ADMIN"
-const POWERUSER_LEVEL_ID = "POWER_USER"
-const BUILDER_LEVEL_ID = "BUILDER"
-const ANON_LEVEL_ID = "ANON"
-
 // Permissions
-const READ_MODEL = "read-model"
-const WRITE_MODEL = "write-model"
-const READ_VIEW = "read-view"
-const EXECUTE_WORKFLOW = "execute-workflow"
-const USER_MANAGEMENT = "user-management"
-const BUILDER = "builder"
-const LIST_USERS = "list-users"
-
-const adminPermissions = [
+module.exports.READ_MODEL = "read-model"
+module.exports.WRITE_MODEL = "write-model"
+module.exports.READ_VIEW = "read-view"
+module.exports.EXECUTE_WORKFLOW = "execute-workflow"
+module.exports.USER_MANAGEMENT = "user-management"
+module.exports.BUILDER = "builder"
+module.exports.LIST_USERS = "list-users"
+// Access Level IDs
+module.exports.ADMIN_LEVEL_ID = "ADMIN"
```

Comment:
NAB: You can use an object here to export everything.

```js
 Module.exports = {
   LIST_USERS: “list-users”,
  ..... everything else
}
```

Reply:
I actually dis-like this method of exporting because it gets into the scenario (sometimes) that access levels was previously in, which is that it can cause circular dependencies.

When the `require` statement is first called the module is configured with its own `module.export` object, which can be mutated to add the properties onto it in the method that I have used.

If you define a new object rather than mutating the existing one you can enter into some pretty nasty circular dependency issues, ones that should be resolved, but there was actually on in the system that I uncovered when I moved some files around where the utility access levels were dependent on the workflow controllers, which unless it was handled this way the access levels module was simply exporting its standard, un-mutated object.

Reply:
Ah, makes total sense in that case. Yeah I would eventually like to look at typescript potentially for our server side code or even just run babel over it so we can get some nice bits like ESM.

Reply:
Yeap running Babel over the top is a good solution, means you can use stuff like [babel root plugins](https://github.com/entwicklerstube/babel-plugin-root-import) as well which can really help in multi-layered projects!